### ISSUE TITLE
request: FieldHub API endpoint for project creation

### ISSUE THREAD

**Issue by chris-jan-trapp** [dainst/idai-field#115]
For our use case an endpoint that either replicates the CLI command to create a project and user, taking a project name and a password as parameters, would make sense. Even nicer would be one that takes just the project name and returns a generated password.

Alternatively a method to create a project's image directories from outside the container would be useful (but feel hacky).

**Comment by dersmon** [comment]
For testing you can now download https://github.com/dainst/idai-field/pkgs/container/field_hub/62455265?tag=3.1.4-rc1. 

You should be able to create projects with a POST on `/projects/<your project name>`. You can send a JSON payload to pre-define a password:

```
{
    "password": "<your password>"
}
```

A user with the same name as the project should get added. The response:

```
{
    "info": {
        "password": "password",
        "status_project": {
            "database": "created",
            "file_store": {
                "original_image": "ok",
                "thumbnail_image": "ok"
            }
        },
        "status_role": "set",
        "status_user": "created"
    }
}
```

**Comment by dersmon** [comment]
Ah, you need to run the `FieldHub.CLI.setup()` once.

**Comment by dersmon** [comment]
Route changed to `/project`. Will return 412 status if either the project or the default user of the same name already exist.

**Comment by dersmon** [comment]
Feature will be released in 3.2.0 of FieldHub.
